lib/msf/core/payload_set.rb

```diff
@@ -70,140 +70,226 @@ def recalculate
     new_keys = []
 
     # Recalculate single payloads
-    _singles.each_pair { |name, op|
-      mod, handler = op
-
-      # if the payload has a dependency, check
-      # if it is supported on the system
-      payload_dependencies = op[4].dependencies
-      unless payload_dependencies.empty?
-        supported = payload_dependencies.all?(&:available?)
-        elog("Dependency for #{name} is not supported") unless supported
-        next unless supported
-      end
-
-      # Build the payload dupe using the determined handler
-      # and module
-      p = build_payload(handler, mod)
+    _singles.each_pair do |name, op|
+      single_payload = calculate_single(name: name, single_components: op)
+      next unless single_payload
 
-      # Add it to the set
-      add_single(p, name, op[5])
       new_keys.push name
 
-      _adapters.each_pair { |adapter_name, ep|
-        adapter_mod, _, adapter_platform, adapter_arch, adapter_inst = ep
-        next unless adapter_inst.compatible?(p.new)
-
-        ap = build_payload(handler, mod, adapter_mod)
+      _adapters.each_pair do |adapter_name, ep|
+        adapted_single = calculate_adapted_single(adapter_name: adapter_name,
```

Comment:
Would this be clearer with adding a `_payload` suffix?

```suggestion
        adapted_single_payload = calculate_adapted_single_payload(adapter_name: adapter_name,
```

Reply:
I'm not against it, but this is consistent with how I did the rest, would you want them all to have `_payload` at the end?